tests/history-missing-bug.rs

```diff
@@ -0,0 +1,168 @@
+    let mut wlt_2 = get_wallet(DescriptorType::Wpkh);
+
+    println!("Wallets created: wlt_1 and wlt_2");
+
+    let amount = 100000;
+    println!("Initial amount: {}", amount);
+
+    let (contract_id, iface_type_name) = wlt_1.issue_nia(amount, wlt_1.close_method(), None);
+    println!("Contract issued - ID: {:?}, Interface type: {}", contract_id, iface_type_name);
+
+    let initial_height = get_height();
+    println!("Initial height: {}", initial_height);
+
+    let print_history = |history: HashMap<XWitnessId, IfaceOp<AmountChange>>| {
+        println!("Amount\tCounterparty\tWitness Id");
+        for (id, op) in history {
+            let (amount, cparty, more) = match op.state_change {
+                AmountChange::Dec(amt) => (
+                    format!("-{}", amt.value()),
+                    op.beneficiaries.first(),
+                    op.beneficiaries.len().saturating_sub(1),
+                ),
+                AmountChange::Zero => continue,
+                AmountChange::Inc(amt) => (
+                    format!("{}", amt.value()),
+                    op.payers.first(),
+                    op.payers.len().saturating_sub(1),
+                ),
+            };
+            let more = if more > 0 { format!(" (+{more})") } else { s!("") };
+            let cparty = cparty
+                .map(XOutputSeal::to_string)
+                .unwrap_or_else(|| s!("none"));
+            println!("{},{}\t{}{}\t{}", amount, op.state_change, cparty, more, id);
+        }
+    };
+
+    let history = wlt_1.raw_rgb_wallet().fungible_history(contract_id, iface_type_name.clone()).unwrap();
+    print_history(history);
+
+    let amount_transfer = 200;
+    println!("Transfer amount: {}", amount_transfer);
+
+    let invoice = wlt_2.invoice(
+        contract_id,
+        &iface_type_name,
+        amount_transfer,
+        wlt_2.close_method(),
+        InvoiceType::Witness,
+    );
+    println!("Invoice created: {:?}", invoice);
+
+    let (consignment, _) = wlt_1.transfer(invoice.clone(), None, Some(1234));
+
+    let height = get_height();
+    println!("Height: {}", height);
+
+    wlt_2.accept_transfer(consignment);
+
+    mine(false);
+    wlt_1.sync();
+    wlt_2.sync();
+
+    let height = get_height();
+    println!("Height: {}", height);
+
+    let history = wlt_1.raw_rgb_wallet().fungible_history(contract_id, iface_type_name.clone()).unwrap();
+    println!("History 1:");
+    print_history(history);
+
+    println!("Balance 1:");
+    let balance = wlt_1.debug_logs(contract_id, &iface_type_name.clone());
+    println!("{:?}", balance);
+
+    let history = wlt_2.raw_rgb_wallet().fungible_history(contract_id, iface_type_name.clone()).unwrap();
+    println!("History 2:");
+    print_history(history);
+
+    println!("Balance 2:");
+    let balance = wlt_2.debug_logs(contract_id, &iface_type_name.clone());
+    println!("{:?}", balance);
+}
+
+#[test]
+fn self_transfer_example() {
```

Comment:
please rename this to `send_to_oneself` and instead of just printing add some asserts to make the test fail